toolkit/scripts/license_map.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/python3
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import argparse
+from collections import OrderedDict
+import json
+from os.path import isdir, isfile
+from pathlib import Path
+import sys
+
+from spec_source_attributions import get_spec_source, KNOWN_SOURCE_ORIGINS
+
+# Packages with specs not present in any "SPECS*" directory, which still need to be included in the list.
+spec_exceptions = {
+    "Microsoft": {
+        "kubernetes-1.18.14",
+        "kubernetes-1.18.17",
+        "kubernetes-1.19.7",
+        "kubernetes-1.19.9",
+        "kubernetes-1.20.2",
+        "kubernetes-1.20.5"
+        }
+    }
+
+# Expected Schema:
+# class LicenseCollection:
+#     header: str
+#     table_headers: List[str]
+#     licenses: List[License]
+#
+# class License:
+#     origin: str
+#     license: str
+#     specs: List[str]
+
+def generate_markdown(license_collection):
+    res = []
+    res.append(license_collection["header"])
+    res.append('')
+    res.append('| ' + ' | '.join(license_collection["table_headers"]) + ' |')
+    res.append('|' + '|'.join([' --- ' for _ in license_collection["table_headers"]]) + '|')
+
+    for origin, details in license_collection["licenses"].items():
+        details["specs"].sort(key=str.lower)
+        res.append('| {0} | {1} | {2} |'.format(origin, details["license"], ' <br> '.join(details["specs"])))
+    
+    return '\n'.join(res) + '\n'
+
+
+def sort_licenses(license_collection):
+    license_collection["licenses"] = OrderedDict(sorted(license_collection["licenses"].items(), key=lambda item:str.lower(item[0])))
+    for details in license_collection["licenses"].values():
+        # Remove duplicates with "set()" and return a sorted list.
+        details["specs"] = sorted(set(details["specs"]), key=str.lower)
+
+
+def process_file(spec_path, license_collection, specs_in_files, specs_unknown_distro):
```

Comment:
Nit: Would be nice for the function name to specify which files are being processed.

Reply:
Most recent change looks good, thanks!